packages/web/components/funnels/concentrated-liquidity/select-cfmm-to-cl-migration.tsx

```diff
@@ -0,0 +1,122 @@
+import { ButtonHTMLAttributes, FunctionComponent } from "react";
+import { useTranslation } from "react-multi-lang";
+
+import { Disableable } from "~/components/types";
+import { ModalBase, ModalBaseProps } from "~/modals";
+import { useStore } from "~/stores";
+
+import { useCfmmToClMigration } from "./use-cfmm-to-cl-migration";
+
+export const SelectCffmToClMigration: FunctionComponent<
+  { cfmmPoolId: string; onSuccessfulMigrate: () => void } & ModalBaseProps
+> = observer((props) => {
+  const { cfmmPoolId } = props;
+  const t = useTranslation();
+
+  const {
+    derivedDataStore,
+    accountStore,
+    chainStore: {
+      osmosis: { chainId },
+    },
+  } = useStore();
+
+  const account = accountStore.getAccount(chainId);
+
+  const { sharePoolDetail } = derivedDataStore.getForPool(cfmmPoolId);
+
+  const { isLinked, migrate } = useCfmmToClMigration(cfmmPoolId);
+
+  if (!isLinked) return null;
+
+  return (
+    <ModalBase
+      title={t("addConcentratedLiquidityPoolCta.migration.title")}
+      {...props}
+    >
+      <div className="flex flex-col gap-3 p-3">
+        {!sharePoolDetail.userAvailableShares.toDec().isZero() && (
+          <OptionButton
+            disabled={Boolean(account.txTypeInProgress)}
+            title={t(
+              "addConcentratedLiquidityPoolCta.migration.availableSharesTitle"
+            )}
+            description={t(
+              "addConcentratedLiquidityPoolCta.migration.availableSharesDescription"
+            )}
+            onClick={() => {
+              migrate({ cfmmShares: sharePoolDetail.userAvailableShares })
+                .then(() => props.onSuccessfulMigrate())
+                .catch(console.error)
+                .finally(() => props.onRequestClose());
+            }}
+          >
+            <span className="caption ml-auto text-osmoverse-100">
+              {t("addConcentratedLiquidityPoolCta.migration.balance")}{" "}
+              {sharePoolDetail.userAvailableShares.trim(true).toString() ?? ""}{" "}
+              ({sharePoolDetail.userAvailableValue.toString() ?? ""})
+            </span>
+          </OptionButton>
+        )}
+        <OptionButton
+          disabled={
+            sharePoolDetail.userBondedShares.toDec().isZero() ||
+            Boolean(account.txTypeInProgress)
+          }
+          title={t(
+            "addConcentratedLiquidityPoolCta.migration.bondedSharesTitle"
+          )}
+          description={t(
+            "addConcentratedLiquidityPoolCta.migration.bondedSharesDescription"
+          )}
+          onClick={() => {
+            migrate({
+              lockIds: sharePoolDetail.userLockedAssets
+                .flatMap(({ lockIds }) => lockIds)
+                .concat(
+                  sharePoolDetail.userUnlockingAssets.flatMap(
+                    ({ lockIds }) => lockIds
+                  )
+                ),
+            })
+              .then(() => props.onSuccessfulMigrate())
+              .catch(console.error)
```

Comment:
console

Reply:
Yep, errors directed at users are defined in root store, where account store is constructed